include/SugarPHPMailer.php

```diff
@@ -407,4 +407,67 @@ protected function embeddedAttachmentExists($filename)
         return false;
     }
 
+    /**
+     * overloads PHPMailer::Send() to allow for better logging and debugging SMTP issues
+     *
+     * @return bool
+     */
+    public function send() 
+    {
+        //Use these to override some fields for tests when debugging SMTP issues:
+        //$this->Host     = "smtp.myserver.com";
+        //$this->From     = 'me@here.com';
+        //$this->FromName = 'My Name';
+        //$this->Sender   = 'me@here.com';
+        //$this->Password = 'wrong';
+        //$GLOBALS['log']->debug("PHPMailer Send Function: { FromName: $this->FromName From: $this->From Host: $this->Host UserName: $this->Username }");
+
+        $this->fullSmtpLog='';
+        $phpMailerExceptionMsg='';
+
+        try {
+            $saveExceptionsState = $this->exceptions;
+            $this->exceptions = true;
+
+            // pass callabck function for PHPMailer to call to provide log :
+            $this->Debugoutput = function($str, $level) {
+                // obfuscate part of response if previous line was a server 334 request, for authentication data: 
+                static $previousIs334 = false;  
+                if ($previousIs334) {
+                    $this->fullSmtpLog .= "$level: CLIENT -> SERVER: ---obfuscated---\n";
+                } else {
+                    $this->fullSmtpLog .= "$level: $str\n";
+                }
+                $previousIs334 = (strpos($str, 'SERVER -> CLIENT: 334') !== false);
+            };
+
+            $this->SMTPDebug = 3;
+            $ret = parent::send();
+            $this->exceptions =  $saveExceptionsState;
+        }
+        catch (Exception $e) {
+            $phpMailerExceptionMsg=$e->errorMessage(); //Pretty error messages from PHPMailer
+            if ($phpMailerExceptionMsg) {
+                $GLOBALS['log']->error("send: PHPMailer Exception: { $phpMailerExceptionMsg }");
+            }
+        }
+        catch (\Exception $e) { //The leading slash means the Global PHP Exception class will be caught
```

Comment:
class `SugarPHPMailer` has not any namespace so in this context the `Exception` and `\Exception` means same thing, you don't have to handle it separately.

Reply:
Hhmm....  I got this code directly from PHPMailer example here: https://github.com/PHPMailer/PHPMailer/blob/master/examples/exceptions.phps

Reply:
Maybe it's for the next version of PHPMailer, they introduced a namespace if I'm not mistaken. I think we should be updating PHPMailer soon, so we could leave this here if it doesn't hurt. What do you think?

Reply:
I think, in this example they using a class alias `use PHPMailer\PHPMailer\Exception;` for `Exception` here: https://github.com/PHPMailer/PHPMailer/blob/master/examples/exceptions.phps#L8  consequently there the `Exception` are same as `PHPMailer\PHPMailer\Exception` and it is why they have to mark the original base php `\Exception` with root namespace but here in this file `include/SugarPHPMailer.php` doesn't contains anything like this, so in this context the `Exception` and `\Exception` are same. If you want to using the PHPMailer specified Exception separately as it is in the example, you should `use` it in SugarPHPMailer class. 

Because confusing name similarity I prefer something like this here: `use PHPMailer\PHPMailer\Exception as PHPMailerException;` and after you can handle just like in they example: 
```
} catch (PHPMailerException $e) {
    echo $e->errorMessage(); //Pretty error messages from PHPMailer
} catch (Exception $e) { //The leading slash doesn't need anymore
    echo $e->getMessage(); //Boring error messages from anything else!
}
```
if you want to learn more about namespace importing and aliasing, please see:
http://php.net/manual/en/language.namespaces.importing.php

Reply:
And last think: you right, if we are using an older PHPMailer version, unfortunately you should wait with this change unit we upgrade the PHPMailer, because this PR doesn't compatible with current SuiteCRM code.

Reply:
In this case the class being thrown is 'phpmailerException', and it is being caught by the first catch block, making the second one obsolete.